src/org/centos/pipeline/PipelineUtils.groovy

```diff
@@ -160,4 +169,254 @@ def setMessageFields(messageType){
+    stages.get(currentStage).each { key, value ->
+        env."${key}" = value
+    }
+}
+
+/**
+ * Library to create text and write to file based on current stage and calls runTaskAndReturnLogs() which rsyncs
+ * the logs produced from executing a task to artifacts.ci.centos.org
+ *
+ * variables
+ *  currentStage - current stage running
+ */
+def rsyncData(currentStage){
+    def text = "export JENKINS_JOB_NAME=\"${JOB_NAME}-${currentStage}\"\n" +
+            "export RSYNC_USER=\"${RSYNC_USER}\"\n" +
+            "export RSYNC_SERVER=\"${RSYNC_SERVER}\"\n" +
+            "export RSYNC_DIR=\"${RSYNC_DIR}\"\n" +
+            "export FEDORA_PRINCIPAL=\"${FEDORA_PRINCIPAL}\"\n" +
+            "export JENKINS_BUILD_TAG=\"${BUILD_TAG}-${currentStage}\"\n" +
+            "export OSTREE_BRANCH=\"${OSTREE_BRANCH}\"\n"
+
+    if (currentStage in ['ci-pipeline-ostree-compose', 'ci-pipeline-ostree-iamge-compose',
+                         'ci-pipeline-ostree-image-boot-sanity', 'ci-pipeline-ostree-boot-sanity']) {
+        text = text +
+                "export HTTP_BASE=\"${HTTP_BASE}\"\n" +
+                "export branch=\"${branch}\"\n"
+    }
+    if (currentStage == 'ci-pipeline-rpmbuild') {
+        text = text +
+                "export fed_repo=\"${fed_repo}\"\n" +
+                "export fed_branch=\"${fed_branch}\"\n" +
+                "export fed_rev=\"${fed_rev}\"\n"
+
+    } else if (currentStage == 'ci-pipeline-ostree-image-boot-sanity') {
+        text = text +
+                "export ANSIBLE_HOST_KEY_CHECKING=\"False\"\n"
+    } else if (currentStage == 'ci-pipeline-ostree-boot-sanity') {
+        text = text +
+                "export fed_repo=\"${fed_repo}\"\n" +
+                "export image2boot=\"${image2boot}\"\n" +
+                "export commit=\"${commit}\"\n" +
+                "export ANSIBLE_HOST_KEY_CHECKING=\"False\"\n"
+    }
+
+    writeFile file: "${env.ORIGIN_WORKSPACE}/task.env",
+            text: text
+    runTaskAndReturnLogs(currentStage)
+
+}
+
+/**
+ * Library to provision resources used in the current stage
+ *
+ * variables
+ *  currentStage - current stage running
+ */
+def provisionResources(currentStage){
+    def duffyOp = "--allocate"
+    allocDuffy(currentStage, duffyOp)
+
+    echo "Duffy Allocate ran for stage ${currentStage} with option ${duffyOp}\r\n" +
+            "ORIGIN_WORKSPACE=${ORIGIN_WORKSPACE}\r\n" +
+            "ORIGIN_BUILD_TAG=${ORIGIN_BUILD_TAG}\r\n" +
+            "ORIGIN_CLASS=${ORIGIN_CLASS}"
+
+    job_props = "${ORIGIN_WORKSPACE}/job.props"
+    job_props_groovy = "${ORIGIN_WORKSPACE}/job.groovy"
+    convertProps(job_props, job_props_groovy)
+    load(job_props_groovy)
+
+}
+
+/**
+ * Library to teardown resources used in the current stage
+ *
+ * variables
+ *   currentStage - current stage running
+ */
+def teardownResources(currentStage){
+    def duffyOp = "--teardown"
+    allocDuffy(currentStage, duffyOp)
```

Comment:
We should call the cico library

Reply:
are you referring to the teardownDuffyCciskel() method in cico-library?

Reply:
Yes